kitsune/wiki/utils.py

```diff
@@ -119,11 +126,22 @@ def get_featured_articles(product=None, locale=settings.WIKI_DEFAULT_LANGUAGE):
         .exclude(document__is_template=True)
         .order_by("-visits")
         .select_related("document")
+        .annotate(
+            topic_ids=StringAgg(
+                Cast("document__topics__id", TextField()),
+                delimiter=",",
+                ordering="document__topics__id",
+                default="",
+            )
+        )
     )
 
     if product:
         visits = visits.filter(document__products__in=[product.id])
 
+    if topic:
```

Comment:
nit: This section could be optimized for efficiency
```python
filters = {}
if product:
    filters['document__products__in'] = [product.id]
if topic:
    filters['document__topics__in'] = [topic.id]

for filter_name, filter_value in filters.items():
    visits = visits.filter(**{filter_name: filter_value})
```

Reply:
Thanks!